integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+	"github.com/grafana/mimir/pkg/storegateway"
+	"github.com/grafana/mimir/pkg/util/globalerror"
+)
+
+const (
+	blockRangePeriod = 5 * time.Second
+	querierTag       = "querier"
+	orgID            = "test"
+	series1          = "series_1"
+	series2          = "series_2"
+	series3          = "series_3"
+	series4          = "series_4"
+)
+
+func Test_MaxSeriesAndChunksPerQueryLimitHit(t *testing.T) {
+	setup := func(t *testing.T, additionalStoreGatewayFlags, additionalQuerierFlags map[string]string) (scenario *e2e.Scenario, ingester, storeGateway *e2emimir.MimirService, client *e2emimir.Client) {
+		scenario, err := e2e.NewScenario(networkName)
+		require.NoError(t, err)
+
+		flags := mergeFlags(
+			BlocksStorageFlags(),
+			BlocksStorageS3Flags(),
+			map[string]string{
+				"-ingester.ring.replication-factor": "1",
+
+				// Frequently compact and ship blocks to storage so we can query them through the store gateway.
+				"-blocks-storage.bucket-store.sync-interval":        "1s",
+				"-blocks-storage.tsdb.block-ranges-period":          blockRangePeriod.String(),
+				"-blocks-storage.tsdb.head-compaction-idle-timeout": "1s",
+				"-blocks-storage.tsdb.retention-period":             ((blockRangePeriod * 2) - 1).String(),
+				"-blocks-storage.tsdb.ship-interval":                "1s",
+			},
+		)
+
+		consul := e2edb.NewConsul()
+		minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+		require.NoError(t, scenario.StartAndWaitReady(consul, minio))
+
+		// Start Mimir components.
+		distributor := e2emimir.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flags)
+
+		ingester = e2emimir.NewIngester("ingester", consul.NetworkHTTPEndpoint(), flags)
+
+		additionalStoreGatewayFlags = mergeFlags(flags, additionalStoreGatewayFlags)
+		storeGateway = e2emimir.NewStoreGateway("store-gateway", consul.NetworkHTTPEndpoint(), additionalStoreGatewayFlags)
+
+		require.NoError(t, scenario.StartAndWaitReady(distributor, ingester, storeGateway))
+
+		additionalQuerierFlags = mergeFlags(flags, additionalQuerierFlags)
+		querier := e2emimir.NewQuerier(querierTag, consul.NetworkHTTPEndpoint(), additionalQuerierFlags)
```

Comment:
[nit] The `querierTag` is only used here. Can be removed.

Reply:
I missed this one. Sorry for this.